CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -205,6 +205,12 @@ public static function cache($userID, $type = CRM_Core_Permission::VIEW, $force
         CRM_Core_Permission::EDIT => [],
       ];
     }
+    // We may find ourselves here if a function has called CRM_ACL_BAO_Cache::updateEntry function. That function doesn't appropriately
+    // Check for view all contacts permission or similar which we don't need to cache.
+    if (CRM_Core_Permission::check('edit all contacts') || $type == CRM_ACL_API::VIEW && CRM_Core_Permission::check('view all contacts')) {
+      Civi::$statics[__CLASS__]['processed'][$type][$userID] = 1;
```

Comment:
Actually this is not quite right - see the next function - it would work for edit all & delete all 

```
    $deleteClause = ' ( 1 ) ';
    if (!$skipDeleteClause) {
      if (CRM_Core_Permission::check('access deleted contacts') and $onlyDeleted) {
        $deleteClause = '(contact_a.is_deleted)';
      }
      else {
        // CRM-6181
        $deleteClause = '(contact_a.is_deleted = 0)';
      }
    }

    if (!$contactID) {
      $contactID = CRM_Core_Session::getLoggedInContactID();
    }
    $contactID = (int) $contactID;

    // first see if the contact has edit / view all permission
    if (CRM_Core_Permission::check('edit all contacts', $contactID) ||
      ($type == self::VIEW && CRM_Core_Permission::check('view all contacts', $contactID))
    ) {
      return $deleteClause;
    }
```

Reply:
@eileenmcnaughton So this function is only about populating the civicrm_acl_contact_cache table, Where as that function you quote is about actually generating the relevant where clause for joins in reports etc

Reply:
which you can see in that function it is returning a where clause without consideration of the acl_contact_cache table

Reply:
@seamuslee001 that function is called from the function you are editing to populate civicrm_acl_contact_cache - we shouldn't need the cache at all when ALL contacts can be viewed - but you can only view all with edit + access deleted (or view + access deleted)

```
    $permission = CRM_ACL_API::whereClause($type, $tables, $whereTables, $userID, FALSE, FALSE, TRUE);

    $from = CRM_Contact_BAO_Query::fromClause($whereTables);
    CRM_Core_DAO::executeQuery("
INSERT INTO civicrm_acl_contact_cache ( user_id, contact_id, operation )
SELECT DISTINCT $userID as user_id, contact_a.id as contact_id, '{$operation}' as operation
         $from
         LEFT JOIN civicrm_acl_contact_cache ac ON ac.user_id = $userID AND ac.contact_id = contact_a.id AND ac.operation = '{$operation}'
WHERE    $permission
AND ac.user_id IS NULL
");
```